torch/csrc/jit/fuser/codegen.cpp

```diff
@@ -456,10 +446,34 @@ std::string generateKernel(
       AT_ASSERT(use_cuda);
       has_random = true;
     }
+    // Always emit double for prim::Constant. This will be narrowed later based
+    // on either:
+    //  - Tensor-Scalar operator type rules
+    //  - Math function rules
+    if (n->kind() == prim::Constant) {
+      const auto val = toIValue(n->output()).value();
+      std::string rhs;
+      if (val.isDouble()) {
+        rhs = scalarValue(val.toDouble());
+      } else if (val.isBool()) {
+        rhs = scalarValue(val.toBool());
+      } else {
+        AT_ASSERT(val.isInt());
+        rhs = scalarValue(val.toInt());
+      }
+      env.s("node", valueName(n->output()));
+      env.s("rhs", rhs);
+      auto var_type = variableType(n->output()->type());
```

Comment:
variableType is wrong. It should return 'int64_t' for IntType and 'double' for FloatType. There should be no need for the if statement here.

Reply:
Somebody made it so that the interface to fused kernels that take scalar floating point arguments is `float`, probably because the fusion compiler is biased toward GPU code. I can change that interface, but it would involve mucking with the GPU fuser